pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/SnapshotSegmentAbortedTxnProcessorImpl.java

```diff
@@ -760,11 +763,13 @@ private CompletableFuture<Void> clearAllSnapshotSegments() {
            });
         }
 
-
-        CompletableFuture<Void> closeAsync() {
-            return CompletableFuture.allOf(
-                    this.snapshotIndexWriterFuture.thenCompose(SystemTopicClient.Writer::closeAsync),
-                    this.snapshotSegmentsWriterFuture.thenCompose(SystemTopicClient.Writer::closeAsync));
+        synchronized CompletableFuture<Void> closeAsync() {
+            if (!closed) {
+                closed = true;
+                this.snapshotIndexWriterFuture.close();
+                this.snapshotSegmentsWriterFuture.close();
```

Comment:
We should call `release` instead of `close` here, right?

Reply:
fixed